### ISSUE TITLE
Engineer features in a globally accepted format

### ISSUE THREAD

**Issue by andreassot10** [CDU-data-science-team/pxtextmining#22]
Different classifiers may perform better with [normalized data](https://scikit-learn.org/stable/modules/sgd.html#tips-on-practical-use) or may have strict non-negativity requirements or format requirements (e.g. `sklearn.naive_bayes.CategoricalNB` is for categorical data only). Moreover, feature selection with `sklearn.feature_selection.chi2` assumes features are counts, boolean, frequencies etc.

[So far](https://github.com/CDU-data-science-team/pxtextmining/blob/7768a6ee7113d6c4401d903026498ed96decbf9f/pxtextmining/factories/factory_pipeline.py#L71-L87), the features are TF-IDFs, plus a few `TextBlob` and `VADER` indicators that are not counts or TF-IDFs, and some of them can also be negative. A categorical feature about the question context will also be added at some point (#23). 

A "global" feature format is required here, probably this one:
- TF-IDFs.
- One-hot encoded categorical variables.
- Continuous variables (not TF-IDFs) converted into categorical bins and then also one-hot encoded.

**Comment by andreassot10** [comment]
Just thought that a different pre-processing is required for embeddings.
- When embeddings are used, feature selection is performed with `sklearn.feature_selection.f_classif` instead of `sklearn.feature_selection.chi2`. Better scale the data to have zero mean and unit variance;
- Some models are not scale-invariant (e.g. SGD).  Again better scale the data to have zero mean and unit variance;
- Contrary to the model with TF-IDF (that uses `sklearn.feature_selection.chi2`), the pre-processing of the continuous sentiment indicators does not require discretization in this case- just scaling to have zero mean and unit variance to satisfy `sklearn.feature_selection.f_classif` and the scale-invariant models.

We can use `pxtextmining.helpers.scaler_switcher.ScalerSwitcher()` with `scaler=KBinsDiscretizer(encode='onehot')` for TF-IDF and with `scaler=StandardScaler()` for embeddings.
